modin/data_management/partitioning/partition_collections.py

```diff
@@ -119,9 +147,21 @@ def block_widths(self):
             # The first column will have the correct lengths. We have an
             # invariant that requires that all blocks be the same width in a
             # column of blocks.
-            self._widths_cache = [obj.width().get() for obj in self.partitions[0]]
+            self._widths_cache = (
+                [obj.width().get() for obj in self._partitions_cache[0]]
+                if len(self._partitions_cache) > 0
+                else []
+            )
         return self._widths_cache
 
+    def _remove_empty_blocks(self):
+        if self._widths_cache is not None:
```

Comment:
Why does nothing happen when `self._widths_cache` is `None`?

Reply:
I think it's to avoid iterating through a NoneType object in the list comprehension in the next line.

Reply:
By default `_widths_cache` is `None` to avoid computing on the remotes without need. We don't need to change the contents of `_widths_cache` if nothing is there yet, and we don't want to trigger the compute here either.